torch/csrc/jit/source_range.h

```diff
@@ -79,6 +119,25 @@ struct Source {
     return serialized_;
   }
 
+  static std::shared_ptr<Source> __setstate__(const c10::IValue& iv) {
+    auto tup_elems = iv.toTuple()->elements();
+    TORCH_INTERNAL_ASSERT(tup_elems.size() == 3);
+    std::string text_ = tup_elems[0].toString()->string();
+    c10::optional<std::string> filename_ =
+        tup_elems[1].toOptional<std::string>();
+    int64_t starting_line_no_ = tup_elems[2].toInt();
+
+    return std::make_shared<Source>(
+        std::move(text_), std::move(filename_), starting_line_no_);
+  }
+
+  std::shared_ptr<SourceRange> query_debug_info(const SourceRange& sr) const {
```

Comment:
The name of this function doesn't make sense to me. It gives me no idea what it is actually doing. Maybe something like: `findSourceRangeThatGenerated(`